scripts/fetch-cloudwatch-metrics.py

```diff
@@ -0,0 +1,84 @@
+import boto3
+import sys
+import argparse
+from botocore.exceptions import ClientError
+from datetime import datetime, timedelta
+
+
+def generate_metric_query(hosted_zone_id, day_offset, region):
+
+    # Creates a cloudwatch client for a specified region.
+    client = boto3.client('cloudwatch', region_name=region)
+
+    start_time = (datetime.now() - timedelta(days=day_offset)
+                  ).strftime("%Y-%m-%d")
+    end_time = (datetime.now()).strftime("%Y-%m-%d")
+
+    try:
+        response = client.get_metric_data(
+            MetricDataQueries=[
+                {
+                    'Id': 'my_dns_query',
+                    'MetricStat': {
+                        'Metric': {
+                            'Namespace': 'AWS/Route53',
+                            'MetricName': 'DNSQueries',
+                            'Dimensions': [
+                                {
+                                    'Name': 'HostedZoneId',
+                                    'Value': hosted_zone_id
+                                },
+                            ]
+                        },
+                        'Period': 86400,
+                        'Stat': 'Sum',
+                    },
+                    'ReturnData': True,
+                },
+            ],
+            StartTime=start_time,
+            EndTime=end_time
+        )
+    except (ClientError):
+        sys.exit(
```

Comment:
Just been doing [some reading](https://www.scaler.com/topics/exit-in-python/) around `sys.exit()` 👀 

Sounds like it's used to try and stop termination and sneak past exception handlers (though it can still be caught).

Probably not a big concern for this script, but I believe typically it may be better to raise a `ValueError` or some `Exception` so callers have the opportunity to handle this scenario if they want? 🤔

Reply:
Will leave this one as is for now, as it's just acting as a reminder to refresh tokens